WeScan/ImageScannerController.swift

```diff
@@ -62,10 +73,30 @@ public final class ImageScannerController: UINavigationController {
         fatalError("init(coder:) has not been implemented")
     }
     
+    private func setupConstraints() {
+        let blackFlashViewConstraints = [
+            blackFlashView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            blackFlashView.topAnchor.constraint(equalTo: view.topAnchor),
+            view.trailingAnchor.constraint(equalTo: blackFlashView.trailingAnchor),
+            view.bottomAnchor.constraint(equalTo: blackFlashView.bottomAnchor)
+        ]
+        
+        NSLayoutConstraint.activate(blackFlashViewConstraints)
+    }
+    
     override public var supportedInterfaceOrientations: UIInterfaceOrientationMask {
         return .portrait
     }
     
+    internal func flashToBlack() {
+        self.view.bringSubview(toFront: blackFlashView)
+        blackFlashView.isHidden = false
+        let flashDuration = DispatchTime.now() + 0.05
+        DispatchQueue.main.asyncAfter(deadline: flashDuration) {
```

Comment:
self is referenced strongly here

Reply:
Isn't it fine since the block is guaranteed to be executed, at which point self is released?